src/include/snapper/dialogs.rb

```diff
@@ -524,20 +306,22 @@ def SummaryDialog
       Wizard.HideAbortButton
 
       UI.SetFocus(Id(:snapshots_table))
-      enable_snapshot_buttons(!snapshot_items.empty?)
-      UI.ChangeWidget(
-        Id(:configs),
-        :Enabled,
-	configs.size > 1
-      )
+      selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+      enable_buttons([:show, :modify], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
+      enable_buttons([:configs], Snapper.configs.size > 1)
 
       ret = nil
       while true
         ret = UI.UserInput
 
-        selected = Convert.to_integer(
-          UI.QueryWidget(Id(:snapshots_table), :CurrentItem)
-        )
+        selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+
+        # It not should be possible but defensive checking
+        if (ret == :modify || ret == :show) and not selected.size == 1
```

Comment:
`not` and `and` is used only for work-flow statements, which is not this case so please use `&& !`.

see http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/